packages/react/src/react-chat.ts

```diff
@@ -1,51 +1,25 @@
-import { StandardSchemaV1, Validator } from '@ai-sdk/provider-utils';
 import {
   AbstractChat,
+  AbstractChatInit,
   ChatState,
   ChatStatus,
-  ChatTransport,
   InferUIDataParts,
   UIDataPartSchemas,
   UIDataTypes,
   UIMessage,
-  UseChatOptions,
 } from 'ai';
 
 class ReactChatState<MESSAGE_METADATA, DATA_TYPES extends UIDataTypes>
   implements ChatState<MESSAGE_METADATA, DATA_TYPES>
 {
-  private messages: UIMessage<MESSAGE_METADATA, DATA_TYPES>[];
-  private status: ChatStatus = 'ready';
-  private error: Error | undefined = undefined;
+  messages: UIMessage<MESSAGE_METADATA, DATA_TYPES>[];
+  status: ChatStatus = 'ready';
+  error: Error | undefined = undefined;
```

Comment:
with the new signature of `ChatState`, these can just be public properties -- and if an adapter needs to run "middleware" on them, they can declare them as getters and setters (this is what Svelte does after it's compiled)